packages/integration-sdk-core/src/types/step.ts

```diff
@@ -97,6 +100,21 @@ export interface StepGraphObjectMetadataProperties {
    * used to generate documentation.
    */
   relationships: StepRelationshipMetadata[];
+
+  /**
+   * Indicates the graph object types the step ingests that are always partial
+   * datasets.
+   *
+   * This is useful in steps that ingest an ever growing list of entities and
+   * their relationships, ensuring the synchronization system does not delete
+   * older data that will not be seen as ingestion progresses through the list
+   * over numerous executions. For example:
+   *
+   * * Vulnerability findings
+   * * SCM pull requests
+   * * Ticket systems
+   */
+  partialDatasets?: PartialDatasets[];
```

Comment:
I'm actually wondering if this information should be an optional property on `StepEntityMetadata` and `StepRelationshipsMetadata`. That way, this information can easily make its way into automated documentation in the future.

Reply:
I think this is a great idea. I'll take a stab at reworking it....